service/history/task/transfer_task_executor_base.go

```diff
@@ -50,6 +51,8 @@ const (
 	defaultDomainName              = "defaultDomainName"
 )
 
+var nonAlphanumericRegex = regexp.MustCompile(`[^a-zA-Z0-9]+`)
```

Comment:
it's better to not spread these visibility store constraints across the codebase. can you move `sanitizedHeaderKey` to common folder or somewhere in visibility folders?

Reply:
make sense. Now it's in the common folder